### PR TITLE
helper-cli: Add a command for merging repository configurations

### PR DESCRIPTION
**Pull request by fviernau**

### CODE REVIEW

File: helper-cli/src/main/kotlin/common/Utils.kt [hunk 1]
```diff
@@ -630,3 +609,24 @@ private data class LicenseFindingCurationHashKey(
 
 private fun LicenseFindingCuration.hashKey() =
     LicenseFindingCurationHashKey(path, startLines, lineCount, detectedLicense, concludedLicense)
+
```
Comment (mnonnenmacher) [inline helper-cli/src/main/kotlin/common/Utils.kt]: Commit message: Fix "re-establishealphabetical"

File: helper-cli/src/main/kotlin/common/Utils.kt [hunk 2]
```diff
@@ -37,12 +37,16 @@ import com.here.ort.model.Severity
...
+import com.here.ort.model.config.Curations
+import com.here.ort.model.config.Excludes
 import com.here.ort.model.config.IssueResolution
 import com.here.ort.model.config.LicenseFindingCuration
 import com.here.ort.model.config.PathExclude
 import com.here.ort.model.config.RepositoryConfiguration
+import com.here.ort.model.config.Resolutions
 import com.here.ort.model.config.RuleViolationResolution
 import com.here.ort.model.config.ScopeExclude
+
```
Comment (mnonnenmacher) [inline helper-cli/src/main/kotlin/common/Utils.kt]: Remove empty line.

File: helper-cli/src/main/kotlin/common/Utils.kt [hunk 3]
```diff
@@ -630,3 +648,50 @@ internal fun Collection<PathExclude>.mergePathExcludes(
...
+    return result.values.toList()
+}
+
+/**
+ * Merge the given [RepositoryConfiguration] replacing entries with equal matchers.
+ */
+internal fun RepositoryConfiguration.merge(
+    other: RepositoryConfiguration
+) = RepositoryConfiguration(
```
Comment (mnonnenmacher) [inline helper-cli/src/main/kotlin/common/Utils.kt]: Add a line break after `=`, consider adding a return type.

File: helper-cli/src/main/kotlin/commands/MergeRepositoryConfigurationsCommand.kt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import com.here.ort.model.config.RepositoryConfiguration
+import com.here.ort.model.readValue
+import com.here.ort.utils.PARAMETER_ORDER_MANDATORY
+import com.here.ort.utils.expandTilde
+
+import java.io.File
+
+@Parameters(
+    commandNames = ["merge-repository-configurations"],
+    commandDescription = "Merges the given list of input repository configurations files and writes the result to " +
```
Comment (mnonnenmacher) [inline helper-cli/src/main/kotlin/commands/MergeRepositoryConfigurationsCommand.kt]: configuration**~~s~~**

File: helper-cli/src/main/kotlin/common/Utils.kt (hunk 3 above)
Comment (fviernau) [reply helper-cli/src/main/kotlin/common/Utils.kt]: Done.

File: helper-cli/src/main/kotlin/common/Utils.kt (hunk 1 above)
Comment (fviernau) [reply helper-cli/src/main/kotlin/common/Utils.kt]: Done.

File: helper-cli/src/main/kotlin/common/Utils.kt (hunk 2 above)
Comment (fviernau) [reply helper-cli/src/main/kotlin/common/Utils.kt]: Done.

File: helper-cli/src/main/kotlin/commands/MergeRepositoryConfigurationsCommand.kt (hunk 4 above)
Comment (fviernau) [reply helper-cli/src/main/kotlin/commands/MergeRepositoryConfigurationsCommand.kt]: Done.

**Pull request merged**